velox/dwio/parquet/reader/PageReader.cpp

```diff
@@ -104,8 +107,9 @@ PageHeader PageReader::readPageHeader(int64_t remainingSize) {
         bufferStart_,
         bufferEnd_);
 
-    transport = std::make_shared<thrift::ThriftBufferedTransport>(
-        copy, sizeof(PageHeader));
+    // We do Not enforce the Thrift Parquet Page Header size check for V1.
+    transport =
+        std::make_shared<thrift::ThriftBufferedTransport>(copy, 0, false);
```

Comment:
We will have problem in this case because `copy` does not contain the whole header.  We need to get the header size from Thrift protocol and use it to prefetch data.

Reply:
If it is too hard to get the size for now, I would suggest leave the check there, and `setDirectorySizeGuess` in `Options` to a very large size to work around it for now.